datasets/simulated.py

```diff
@@ -41,9 +41,9 @@ def get_data(self):
             np.sum([
                 np.convolve(theta_k, d_k, mode="full")
                 for theta_k, d_k in zip(theta[:, :, i], D)
-            ], axis=0).reshape(-1, 1)
+            ], axis=0).reshape(1, -1)
             for i in range(theta.shape[2])
-        ], axis=1)
+        ], axis=0)
```

Comment:
I feel that this shape is not consistent with the shape of `theta`.
Basically, the "batch_size" is here first while it it last in `theta`.

Could you use the shape `n_samples, n_times` for `y` and `n_samples, n_atoms, n_times_valid` for `theta`?
This will also make the iteration on theta and the convolution more efficient as it will happen on contiguous arrays:
```python
        y = np.array([
            np.sum([
                np.convolve(theta_ik, d_k, mode="full")
                for theta_ik, d_k in zip(theta_i, D)
            ], axis=0)
            for theta_i in theta
        ])